### ISSUE TITLE
Error detected while processing FileType Autocommands for "*": / Error executing lua callback: ...nvim-treesitter/highlight.lua:19: attempt to call field 'start' (a nil value)

### ISSUE THREAD

**Issue by PIesPnuema** [nvim-treesitter/nvim-treesitter#4369]
### Describe the bug

# Full Error:
```
Error detected while processing FileType Autocommands for "*":
Error executing lua callback: .../start/nvim-treesitter/lua/nvim-treesitter/highlight.lua:19: attempt to call field 'start' (a nil value)
stack traceback:
        .../start/nvim-treesitter/lua/nvim-treesitter/highlight.lua:19: in function 'attach'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:505: in function 'attach_module'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:528: in function 'reattach_module'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:131: in function <...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:130>
```
Here is **line 19** from highlight.lua
```
  vim.treesitter.start(bufnr, lang)
 ```
### *** ALL THE `vim.` in the file are underlined as incorrect syntax. Is my lua api having a issue or is this a treesitter configuration problem? *** 

I have reinstalled nvim-treesitter so the highlight.lua file is the most recent config in that repo.

here is the line packer uses:
```
 use ('nvim-treesitter/nvim-treesitter', {run = ':TSUpdate'})
```

Thank you for your help in advance. I have attached a link to my nvim config inside "addtional content"

### To Reproduce

``` shell
nvim anyfile.lua 
```
the error will pop up

### Expected behavior

no error from nvim

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - WARNING: `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - OK: `node` found v18.12.1 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## OS Info:
{
  machine = "x86_64",
  release = "6.0.12-76060006-generic",
  sysname = "Linux",
  version = "#202212290932~1674139725~22.04~ca93ccf SMP PREEMPT_DYNAMIC Thu J"
}

## Parser/Features         H L F I J
  - c_sharp             ✓ ✓ ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ . ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - help                ✓ . . . ✓
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - gitcommit           ✓ . . . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - java                ✓ ✓ . ✓ ✓
  - cmake               ✓ . ✓ . .
  - html                ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - scss                ✓ . ✓ ✓ .
  - vim                 ✓ ✓ ✓ . ✓
  - make                ✓ . ✓ . ✓
  - css                 ✓ . ✓ ✓ ✓
  - bash                ✓ ✓ ✓ . ✓
  - arduino             ✓ ✓ ✓ ✓ ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - http                ✓ . . . ✓
  - javascript          ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by team+vim@tracker.debian.org

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

### Additional context

[here](https://github.com/PIesPnuema/nvim_config) is my entire config for lua and nvim.

**Comment by clason** [comment]
Your Neovim version is too old. Install 0.8.3.

**Comment by clason** [comment]
Well

**Comment by PIesPnuema** [comment]
# FIXED VIA RECOMENDED UPDATE TO NVIM V 0.8.3.

Thank you, 

Sorry for the inconvenience and  not trying that first! The Ubuntu repo is always behind.

Cheers..
